metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -494,27 +679,47 @@ def get_connection_objects(
                 yield obj
 
     def emit_workbooks(self) -> Iterable[MetadataWorkUnit]:
-        projects = (
-            f"projectNameWithin: {json.dumps(self.config.projects)}"
-            if self.config.projects
-            else ""
-        )
+        if self.tableau_project_registry:
+            project_names: List[str] = [
+                project.name for project in self.tableau_project_registry.values()
+            ]
+            project_names_str: str = json.dumps(project_names)
```

Comment:
do we need to catch here?

Reply:
if self.tableau_project_registry is empty then control wouldn't come here (line 686)